files/static/index.js

```diff
@@ -0,0 +1,340 @@
+    $("#submit-form").classList.add("hidden");
+  },
+
+  showSubmission: function() {
+    $("#checkbox-owner").checked = false;
+    $("#checkbox-subdomains").checked = false;
+    $("#submit").disabled = true;
+
+    $("#submit-form").classList.remove("hidden");
+  },
+
+  onSubmitForm: function() {
+    this.submit(this.domainToSubmit).then(function(issues) {
+      console.log("submit:", issues);
+      $("#submit-result").classList.remove("hidden");
+      if (issues.errors.length == 0) {
+        $("#submit-result").textContent = "Submitted successfully!";
+        // TODO: Now try SSL Labs!
+      } else {
+        $("#submit-result").textContent = "There are errors. Please submit your site again.";
+      }
+    });
+  },
+
+  showWaiting: function(domain) {
+    $("#checking").textContent = "Checking  " + domain;
+
+    $("#result-waiting").classList.remove("hidden");
+  },
+
+  hideWaiting: function() {
+    $("#result-waiting").classList.add("hidden");
+  },
+
+  updateURLHash: function(domain) {
+    console.log(domain);
+    if (domain) {
+      history.replaceState({}, document.title, ".?domain=" + domain);
+    } else {
+      history.replaceState({}, document.title, ".");
+    }
+  },
+
+  onDomainChanged: function() {
+    var domain = this.currentDomain();
+    this.showWaiting(domain);
+    this.clearResults();
+    this.hideSubmission();
+    this.updateURLHash(domain);
+    this.clearTheme()
+
+    if (!domain) {
+      $("#result-waiting").classList.add("hidden");
+      return;
+    }
+
+    this.currentResultsDomain = "";
+
+    Promise.all([
+      this.checkdomain(domain),
+      this.status(domain)
+    ]).then(function(values) {
+      this.handleResults(domain, values[0], values[1]);
+    }.bind(this), function() {
+        // TODO: handle failure better.
+        $("#result").classList.remove("hidden");
+        $("#result-waiting").classList.add("hidden");
+    });
+  },
+
+  onCheckBoxChanged: function() {
+    $("#submit").disabled = !($("#checkbox-owner").checked && $("#checkbox-subdomains").checked);
+  },
+
+  handleResults: function(domain, checkDomainIssues, domainStatus) {
+    console.log("handleResults:", checkDomainIssues, domainStatus);
+
+    if (domain !== this.currentDomain()) {
+      return;
+    }
+    if (this.currentResultsDomain === domain) {
+      return;
+    } else {
+      this.currentResultsDomain = domain;
+    }
+
+    if (domain !== this.currentDomain()) {
+      console.log("Outdated result.");
+      return;
+    }
+
+    this.showResults(domain, checkDomainIssues, domainStatus);
+    this.hideWaiting();
+
+    if (checkDomainIssues.errors.length === 0 && ["unknown", "rejected", "removed"].indexOf(domainStatus.status) != -1) {
+      this.domainToSubmit = domain;
+      this.showSubmission();
+    }
+  },
+
+  domainFromURLParam: function() {
+    var match = window.location.search.match(/^\?domain=([^?&]+)$/);
+    return match ? decodeURIComponent(match[1]) : null;
```

Comment:
That is, in JS I exclusively use `element.textContent` to add text to the DOM (like we do in DevTools).